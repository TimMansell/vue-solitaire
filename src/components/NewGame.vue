<template>
  <Button @click="newGame" data-test="new-game-btn">
    New Game
  </Button>
</template>

<script>
import { mapActions } from 'vuex';
import Button from './Button.vue';

export default {
  name: 'NewGame',
  components: {
    Button,
  },
  methods: {
    ...mapActions(['restartGame', 'initGame']),
    newGame() {
      this.restartGame();
      this.initGame();
    },
  },
};
</script>
