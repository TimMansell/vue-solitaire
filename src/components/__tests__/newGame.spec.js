import { shallowMount } from '@vue/test-utils';
import NewGame from '@/components/NewGame.vue';

describe('NewGame.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(NewGame, {
      computed: {
        game: () => ({
          id: 1,
        }),
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('matches disabled snapshot', () => {
    const wrapper = shallowMount(NewGame, {
      computed: {
        game: () => ({
          id: null,
        }),
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
