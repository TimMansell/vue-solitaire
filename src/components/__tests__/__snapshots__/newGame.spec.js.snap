// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGame.vue matches disabled snapshot 1`] = `
<button-stub data-test="new-game-btn" disabled="true">
  New Game
</button-stub>
`;

exports[`NewGame.vue matches snapshot 1`] = `
<button-stub data-test="new-game-btn">
  New Game
</button-stub>
`;
